### ISSUE TITLE
Dependency issue with version 4.27.1

### ISSUE THREAD

**Issue by arisferyanto** [MenoData/Time4J#651]
Due to wrong dependency version of time4j-i18n in version 4.27.1, my maven project now won't compile. It stuck with error like this:

```
[ERROR] Failed to execute goal on project xxx: Could not resolve dependencies for project yyy: Failed to collect dependencies at zzz -> net.time4j:time4j-range:jar:4.13 -> net.time4j:time4j-i18n:jar:4.27.1 -> net.time4j:time4j-core:jar:[4.28,): No versions available for net.time4j:time4j-core:jar:[4.28,) within specified range -> [Help 1]
```

Even though version 4.27.2 already fixed the wrong dependency (4929151a9584f110e4b95e854a83012355b482af), maven still [iterates through all the matching dependency versions](https://eclipse.googlesource.com/aether/aether-core/+/master/aether-impl/src/main/java/org/eclipse/aether/internal/impl/DefaultDependencyCollector.java#402) and recursively check all the child dependencies. And because time4j-range v4.13 declares its dependency of time4j-i18n as:

```
<dependency>
    <groupId>net.time4j</groupId>
    <artifactId>time4j-core</artifactId>
    <version>[4.13,)</version>
    <scope>compile</scope>
</dependency>
```

it matches with 4.13, 4.14, 4.15, ..., 4.17, 4.17.1, and 4.17.2.

I tried to search if it's possible to exclude 4.17.1 when maven resolve the dependencies, but I couldn't find any solution. And I do not have any control over the project that includes time4j-i18n to upgrade its version. Anybody has workaround for this?

Additionally, can I suggest to use fixed version instead of a range of valid versions? It seems to be a good practice to ensure more consistent builds.

**Comment by arisferyanto** [comment]
Found one workaround: explicitly exclude time4j-i18n from the project and add 4.17.2 as a new dependency.

```
<dependency>
	<groupId>xxx</groupId>
	<artifactId>artifact-that-includes-time4j-range</artifactId>
	<exclusions>
		<exclusion>
			<groupId>net.time4j</groupId>
			<artifactId>time4j-range</artifactId>
		</exclusion>
	</exclusions>
</dependency>
<dependency>
	<groupId>net.time4j</groupId>
	<artifactId>time4j-range</artifactId>
	<version>4.27.2</version>
</dependency>
```

The only thing left now is my suggestion to use a fixed version instead of a range. @MenoData any thought around this?

**Comment by MenoData** [comment]
Yeah, my failure. I had first noticed the dependency error shortly after I had pushed v4.27.1 to maven. And maven does not allow to remove broken versions so the only fix was to release a new version. I see, that having chosen the next version (with the dependency fix) as v4.27.2 was not quite sufficient because the dependency error in i18n-pom-file talks about a non-existing higher version v4.28 (only existed locally for me, so a local build was still possible).

Well I think your proposal to change to fixed versions instead of version ranges makes sense to exclude such inconsistencies in the future. I will prepare a change with next version v4.28.
